newton/tests/test_ik.py

```diff
@@ -254,9 +351,17 @@ class TestIKModes(unittest.TestCase):
     pass
 
 
+# Planar REV-REV convergence
 add_function_test(TestIKModes, "test_convergence_autodiff", test_convergence_autodiff, devices)
 add_function_test(TestIKModes, "test_convergence_analytic", test_convergence_analytic, devices)
 add_function_test(TestIKModes, "test_convergence_mixed", test_convergence_mixed, devices)
+
+# FREE-joint convergence
+add_function_test(TestIKModes, "test_convergence_autodiff_free", test_convergence_autodiff_free, devices)
+add_function_test(TestIKModes, "test_convergence_analytic_free", test_convergence_analytic_free, devices)
+add_function_test(TestIKModes, "test_convergence_mixed_free", test_convergence_mixed_free, devices)
+
```

Comment:
It would be nice to have a test case also for a D6 joint which has >= 2 angular dofs to verify the analytical Jacobian / motion subspace works as expected.